src/datasources/stripe/index.ts

```diff
@@ -60,36 +60,84 @@ export const createStripeProductAndPrice = async ({
     unit_amount: number;
   };
   getStripeClient: () => Stripe;
-}) => {
+}): Promise<string> => {
   const stripeClient = getStripeClient();
-  // check if a number is a float or an integrer
-  const productData = await stripeClient.products.create({
-    id: item.id,
-    name: item.name,
-    active: true, // TODO: Adda way to disable tickets based on ticket status.
-    description: item.description,
-    metadata: item.metadata,
-    default_price_data: {
-      currency: item.currency,
-      // Stripe expects the unit_amount to be an integer, and unit_amount_decimal to be a float
-      // if we want to use decimals. (sellin 29.99 USD), we need to use unit_amount_decimal
-      // instead of unit_amount.
-      ...getUnitAmount(item.unit_amount),
-    },
-    shippable: false,
-  });
-
-  const defaultPrice = productData.default_price;
 
-  if (!defaultPrice) {
-    throw new Error("Stripe product and price could not be created.");
+  try {
+    // Try to retrieve the existing product
+    await stripeClient.products.retrieve(item.id);
+
+    // If the product exists, update it
+    const updatedProduct = await stripeClient.products.update(item.id, {
+      name: item.name,
+      description: item.description,
+      metadata: item.metadata,
+    });
+
+    // Check if we need to update the price
+    const existingPrice = updatedProduct.default_price as Stripe.Price;
+    const unitAmountOfItem = getUnitAmount(item.unit_amount);
+
+    if (
+      !existingPrice ||
+      existingPrice.currency !== item.currency ||
+      ("unit_amount" in unitAmountOfItem &&
```

Comment:
Recomendaría evitar el uso de _Magic_ (Magic strings & Magic numbers)

Referencia: https://dev.to/ruben_alapont/magic-numbers-and-magic-strings-its-time-to-talk-about-it-ci2

Reply:
Eliminé los magic strings.

Aquí ahora utilizo typeof en vez de "in":
https://github.com/JSConfCL/gql_api/blob/8905ee8b502810d8e80a8fe577df559853fde3c0/src/datasources/stripe/index.ts#L90-L97

Y aquí moví el Error Code de Stripe a una constante.
https://github.com/JSConfCL/gql_api/blob/8905ee8b502810d8e80a8fe577df559853fde3c0/src/datasources/stripe/index.ts#L54

Reply:
🔥 Oh nice!!!